mbedtls/src/threading.rs

```diff
@@ -9,8 +9,14 @@
 #[cfg(not(feature = "std"))]
 use alloc_prelude::*;
 
+#[cfg(feature = "spin_threading")]
 extern crate spin;
+#[cfg(feature = "spin_threading")]
 use self::spin::{Mutex, MutexGuard};
+
+#[cfg(feature = "rust_threading")]
+use std::sync::{Mutex, MutexGuard};
```

Comment:
This will fail if a user sets both `spin_threading` and `rust_threading`. Use `cfg(all(feature = "rust_threading", not(feature = "spin_threading")))` and document the precedence in the README